docs/getting-started/installation-and-upgrade/install-upgrade-on-a-kubernetes-cluster/air-gapped-upgrades.md

```diff
@@ -38,6 +38,22 @@ helm upgrade rancher ./rancher-<VERSION>.tgz \
 	--set useBundledSystemChart=true # Use the packaged Rancher system charts
 ```
 
+#### Resolving UPGRADE FAILED Error
+
+If you encounter the error message, `Error: UPGRADE FAILED: "rancher" has no deployed releases`, Rancher might have been  installed via the `helm template` command. The `helm template` command can cause problems with Helm hooks. To successfully upgrade Rancher, use the following commands instead:
+
+```
+helm template rancher ./rancher-<VERSION>.tgz \
+    --no-hooks \ # prevent files for Helm hooks from being generated
+	--namespace cattle-system \
+	--set hostname=<RANCHER.YOURDOMAIN.COM> \
+	--set certmanager.version=<CERTMANAGER_VERSION> \
+	--set rancherImage=<REGISTRY.YOURDOMAIN.COM:PORT>/rancher/rancher \
+	--set systemDefaultRegistry=<REGISTRY.YOURDOMAIN.COM:PORT> \ # Set a default private registry to be used in Rancher
+	--set useBundledSystemChart=true # Use the packaged Rancher system charts
+```
+
```

Comment:
Would also need to include the `kubectl apply` command to deploy the templated manifests after the `helm template` command

Reply:
Could you include the command that needs to be added here? Is it simply `helm apply`?

Reply:
`kubectl -n cattle-system apply -R -f ./rancher` (https://ranchermanager.docs.rancher.com/v2.5/getting-started/installation-and-upgrade/install-upgrade-on-a-kubernetes-cluster/air-gapped-upgrades#apply-the-rendered-templates)

Reply:
This would only works if `--output-dir .` parameter is given to the `helm template` command, otherwise it just outputs the YAML on stdout.

From the same link above, the helm command should look like:

```sh
helm template rancher ./rancher-<VERSION>.tgz --output-dir . \
   ...
```

Reply:
Updated the command in [ef12372](https://github.com/rancher/rancher-docs/pull/1066/commits/ef123728d310df49d6fc9e32e2e0bce737f7dd5f)